### PR TITLE
feat: Add support for Spring Boot and Camunda Run.

### LINKED ISSUE #44: Spring Boot compatible configuration
Many developers use Camunda in a Spring Boot Environment

The current - environment variable + properties file - based config mechanism is not a great fit with the default Spring Boot configuration mechanisms.

Since the Camunda engine itself supports configuration via Spring Boot, it would be nice if the mail connector also supported it.

### PR DESCRIPTION
**Pull request by shaarmann**
- Wrap the connector's configuration in a Bean to be configurable via
  application.yml
- Provide a Camunda Platform 7 plugin that registers the connectors and
  that sets the configuration to the bean

Closes #44

### CODE REVIEW

**Comment by ingorichtsmeier** [comment]
Hi @shaarmann,

here is a configuration for the IDE to format the code consisting to the Camunda 7 Platform: https://github.com/camunda/camunda-bpm-platform/tree/master/settings.

Could you please format the code with this formatter applied (indent is 2 spaces)?

File: pom.xml [hunk 1]
```diff
@@ -23,13 +24,15 @@
     <project.build.sourceEncoding>${encoding}</project.build.sourceEncoding>
     <project.build.resourceEncoding>${encoding}</project.build.resourceEncoding>
     <!-- versions -->
-    <version.camunda>7.12.0</version.camunda>
+    <version.camunda>7.17.0</version.camunda>
+    <version.javamail>1.5.5</version.javamail>
+    <version.springBoot>2.7.2</version.springBoot>
```
Comment (ingorichtsmeier) [inline pom.xml:29]: Spring Boot should be 2.6, as 2.7 is only supported for 7.17.2+.

**Review by ingorichtsmeier — CHANGES_REQUESTED** [review]
I think that the spring boot version should be set to latest supported version, which is 2.6

File: pom.xml (hunk 1 above)
Comment (ingorichtsmeier) [reply pom.xml:29]: @shaarmann and @jonathanlukas, what do you think?

Comment (jonathanlukas) [reply pom.xml:29]: True, I did not properly research version compatibility. Let's set it to 2.6

**Review by ingorichtsmeier — APPROVED** [review]

**Pull request merged**